frontend/src/pages/detail/GenreDetail.tsx

```diff
@@ -37,36 +69,58 @@ export default function GenreDetail() {
     setIsModifiable(false);
   };
 
+  // The navigator to switch pages
+  const navigate = useNavigate();
+
   return (
     <div>
       <div className="grid flex justify-content-center">
+        <Toast ref={toast} />
         <link
           rel="stylesheet"
           href="https://unpkg.com/primeflex@3.1.2/primeflex.css"
         ></link>
-        <div className="col-5">
-          <div className="py-5">
-            <h1 className="p-component p-text-secondary text-5xl text-center text-900 color: var(--surface-800);">
-              Genre Details
-            </h1>
+        <div className="flex col-12 p-0">
+          <div className="flex col-1">
+            <Button
+              type="button"
+              label="Back"
+              icon="pi pi-arrow-left"
+              onClick={() => navigate("/genres")}
+              className="p-button-sm my-auto ml-1"
+            />
+          </div>
+          <div className="pt-2 col-10">
+            {isModifiable ? (
+              <h1 className="p-component p-text-secondary text-5xl text-center text-900 color: var(--surface-800);">
+                Modify Genre
+              </h1>
+            ) : (
+              <h1 className="p-component p-text-secondary text-5xl text-center text-900 color: var(--surface-800);">
+                Genre Details
+              </h1>
+            )}
           </div>
+          <div className="flex col-1">
+            <Button
```

Comment:
Deleting should only be enabled if this genre has no books

Reply:
Genre has numBooks or something that describes this

Reply:
good call right now book_cnt return for genre detail is null for some reason